projects/aws/image-builder/builder/artifacts.go

```diff
@@ -0,0 +1,165 @@
+	}
+
+	buildToolingRepoPath := getBuildToolingPath(cwd)
+	bundles, _, err := b.prepBuildToolingRepo(buildToolingRepoPath)
+	if err != nil {
+		return err
+	}
+	supportedReleaseBranches := GetSupportedReleaseBranches()
+	if !SliceContains(supportedReleaseBranches, b.ReleaseChannel) {
+		cleanup(buildToolingRepoPath)
+		log.Fatalf("release-channel should be one of %v", supportedReleaseBranches)
+	}
+
+	kubeVersion := strings.ReplaceAll(b.ReleaseChannel, "-", ".")
+	log.Printf("Setting Kubernetes Version: %s", kubeVersion)
+
+	// Download eks-a artifacts
+	eksaArtifactsPath := filepath.Join(cwd, eksAnywhereArtifactsDirName)
+	eksdArtifactsPath := filepath.Join(cwd, eksDistroArtifactsDirName)
+	if b.Force {
+		cleanup(eksaArtifactsPath)
+		cleanup(eksdArtifactsPath)
+	}
+	versionBundle, err := b.downloadEKSAArtifacts(eksaArtifactsPath, kubeVersion, bundles)
+	if err != nil {
+		return fmt.Errorf("failed to download eks-a artifacts: %v", err)
+	}
+
+	// Download eks-d artifacts
+	eksDReleaseManifestUrl := versionBundle.EksD.EksDReleaseUrl
+	if err := b.downloadEKSDArtifacts(eksdArtifactsPath, eksDReleaseManifestUrl); err != nil {
+		return fmt.Errorf("failed to download eks-d artifacts: %v", err)
+	}
+	cleanup(buildToolingRepoPath)
+	log.Println("All artifacts were successfully downloaded")
+	log.Printf("Please find EKS-A artifacts under %s and EKS-D artifacts under %s directories\n", eksAnywhereArtifactsDirName, eksDistroArtifactsDirName)
+	return nil
+}
+
+func (b *BuildOptions) downloadEKSAArtifacts(outputPath, kubeVersion string, bundles *releasev1.Bundles) (*releasev1.VersionsBundle, error) {
+	var versionBundle releasev1.VersionsBundle
+	for _, bundle := range bundles.Spec.VersionsBundles {
+		if bundle.KubeVersion == kubeVersion {
+			versionBundle = bundle
+			break
+		}
+	}
+
+	eksDBundle := versionBundle.EksD
+	artifactsUrl := map[string]string{
+		"containerd": eksDBundle.Containerd.URI,
+		"crictl":     eksDBundle.Crictl.URI,
+		"etcdadm":    eksDBundle.Etcdadm.URI,
+	}
+	for artifact, url := range artifactsUrl {
+		log.Printf("Downloading EKS-A Artifact: %s\n", artifact)
+		if err := downloadArtifact(outputPath, url); err != nil {
+			return nil, err
+		}
+	}
+	return &versionBundle, nil
+}
+
+func (b *BuildOptions) downloadEKSDArtifacts(outputPath, manifestUrl string) error {
+	releases := &eksDreleasev1.Release{}
+	manifestData, err := readFileFromURL(manifestUrl)
+	if err != nil {
+		return fmt.Errorf("failed to read manifest file from url: %v", err)
+	}
+	if err = k8syaml.Unmarshal(manifestData, releases); err != nil {
+		return err
+	}
+
+	var fullKubeVersion, eksDBaseUrl string
+	for _, component := range releases.Status.Components {
+		if component.Name == "kubernetes" {
+			fullKubeVersion = component.GitTag
+			log.Printf("Full Kube Version: %s\n", fullKubeVersion)
+
+			for _, asset := range component.Assets {
+				if asset.Name == "bin/linux/amd64/kube-apiserver.tar" {
```

Comment:
i know we do not support building arm images, but can we go ahead and parametrize `amd64`, even if we just hardcode it vs letting the user supply?  One less we have to think about later